### PR TITLE
Flamegraph

### PR DESCRIPTION
**Pull request by jacereda**
Script to generate a clickable flamegraph svg.

This one also builds on top of #768.

Sample output:

[out.zip](https://github.com/carp-lang/Carp/files/4605566/out.zip)

### CODE REVIEW

File: flamegraph.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# This can be useful to see how much time is spent in carp vs clang
+# ./flamegraph.sh ./carp.sh -b <file.carp>
+
+HZ=${HZ:-1000}
+prg=`basename $1`
+svg=`mktemp XXXXXX.svg`
+perf record -F $HZ -a -g -- $*
+perf script | stackcollapse-perf.pl | flamegraph.pl > $svg
+xdg-open $svg
+rm $svg
```
Comment (bjorn3) [inline flamegraph.sh:28]: This will likely remove the svg before the view application had a change to open it.

Comment (jacereda) [reply flamegraph.sh:28]: Oh, right. It seems to work for me though. 

Is there some way to view the svg sinchronously?

Comment (bjorn3) [reply flamegraph.sh:28]: Maybe use `read -p "Press enter to delete the temporary svg file" some_var`?

Comment (hellerve) [reply flamegraph.sh:28]: Does `xdg-open` exit before reading the file?

Comment (jacereda) [reply flamegraph.sh:28]: Not that I like it, but perhaps a `sleep 5` will be less intrusive than prompting...

Comment (bjorn3) [reply flamegraph.sh:28]: > Does xdg-open exit before reading the file?

`xdg-open` runs the program to open a file and then it exits. It doesn't wait for the program to exit.

**Comment by eriksvedang** [comment]
The diff on this one is confusing. Very cool tool though, will merge once rebased.

**Comment by eriksvedang** [comment]
(sorry about the docs, they changed now when we replaced lambda with Fn when printing signature)

**Comment by jacereda** [comment]
Rebased.

**Comment by eriksvedang** [comment]
Thanks!

**Pull request merged**